dm/dm/worker/server_test.go

```diff
@@ -368,6 +368,97 @@ func (t *testServer) TestHandleSourceBoundAfterError(c *C) {
 	}), IsTrue)
 }
 
+func (t *testServer) TestServerQueryValidator(c *C) {
```

Comment:
maybe we can just test `w == nil` case, and let integration test to test the other case since there's no other branch and worker.GetValidateStatus is tested already.